packages/wpcom-checkout/src/checkout-line-items.tsx

```diff
@@ -707,6 +731,29 @@ function IntroductoryOfferCallout( {
 	return <DiscountCallout>{ text }</DiscountCallout>;
 }
 
+function PartnerLogo( {
+	coupon,
+	products,
+}: {
+	coupon: string;
+	products?: ResponseCartProduct[];
+} ): JSX.Element | null {
+	const translate = useTranslate();
+
+	if ( hasPartnerCoupon( coupon, products ) ) {
```

Comment:
Does `coupon` include partner data as well? A switch condition could potentially prevent future confusion of why "IONOS" is being shown for test coupons or newly onboarded partners :)

Reply:
It currently does not. In fact it would be a next step to have that data coming from the coupon API somehow (TBD), so that we could potentially serve different partners.

Reply:
We could use the coupon prefix to determine the partner, like we do in `client/jetpack-connect/controller.js::partnerCouponRedirects`. Or even better; refactor the partner detection logic to be a utility function and share it between the two places 😄

Reply:
Thanks for the suggestions @kallehauge!
I managed to extract the partner coupon detecting logic to a helper file, exported from the wpcom-checkout package to be used in a higher lever (composite-checkout), now there's much less logic to be handled in `checkout-line-items.tsx`.